lib/internal/crypto/diffiehellman.js

```diff
@@ -192,6 +193,28 @@ ECDH.prototype.generateKeys = function generateKeys(encoding, format) {
 };
 
 ECDH.prototype.getPublicKey = function getPublicKey(encoding, format) {
+  var f = getFormat(format);
+  var key = this._handle.getPublicKey(f);
+  encoding = encoding || getDefaultEncoding();
+  return encode(key, encoding);
+};
+
+ECDH.convertKey = function convertKey(key, curve, inEnc, outEnc, format) {
+  const encoding = getDefaultEncoding();
+  inEnc = inEnc || encoding;
+  outEnc = outEnc || encoding;
+  var f = getFormat(format);
+  var convertedKey = _ECDHConvertKey(toBuf(key, inEnc), curve, f);
+  return encode(convertedKey, outEnc);
+};
+
+function encode(buffer, encoding) {
```

Comment:
@bnoordhuis I also added the `encode` function since I noticed that the test for `'buffer'` encoding happens a lot in this file. Should I also change that in the PR or should I submit another PR for that?

Reply:
Thanks, doing it in this PR is fine.